test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_mnist.py

```diff
@@ -118,3 +129,47 @@ def _test_mnist_distributed(
         output = predictor.predict(data)
 
         assert output.shape == (batch_size, 10)
+
+        #  Check for Cloudwatch logs
+        if verify_logs:
+            _check_for_cloudwatch_logs(endpoint_name, sagemaker_session)
+
+def _check_for_cloudwatch_logs(endpoint_name, sagemaker_session):
+    client=sagemaker_session.boto_session.client('logs')
+    log_group_name='/aws/sagemaker/Endpoints/'+endpoint_name
+
+    identify_log_stream = client.describe_log_streams(
+        logGroupName=log_group_name,    
+        orderBy='LastEventTime',
+        descending=True,
+        limit=5
+    )
+
+    try:
+        log_stream_name=identify_log_stream['logStreams'][0]['logStreamName']        
+    except IndexError as e:
+        raise RuntimeError(f"Unable to look up log streams for the log group {log_group_name}") from e
+
+
+    log_events_response = client.get_log_events(
+        logGroupName=log_group_name,
+        logStreamName=log_stream_name,
+        limit=50,
+        startFromHead=True
+    )
+    
+    records_available=bool(log_events_response['events'])    
```

Comment:
Will `log_events_response['events']` be undefined or set to an empty data structure? if undefined, this will fail with key error. You can do something like log_events_response.get('events') which defaults to "None" if the key is not found to avoid this scenario

Reply:
No it will return false if there are no log events but should never return a null as long as there is an associated log stream that was retrieved a line above.